lib/index.js

```diff
@@ -508,9 +509,14 @@ async function run(options = {}) {
         const packages = await previousPromise
         // copy object to prevent share .ncurc options between different packageFile, to prevent unpredictable behavior
         const rcResult = getNcurc({ packageFile })
+        let rcConfig = rcResult && rcResult.config ? rcResult.config : {}
+        if (options.recursive) {
+          // Merge config options
+          rcConfig = mergeOptions(options, rcConfig, { keys: Object.keys(rcConfig) })
```

Comment:
`options` consist of command-line options and config options which are combined (prematurely imo) in `cli.js`. Thus we don't have direct access to the root package config at this point. My previous comment was saying that ideally we only merge config options (not command line options) but that doesn't seem possible without some additional refactoring. Luckily it doesn't seem to be a problem with the main use case we are solving here.

So, I may not have been following the `key` selection logic correctly, but I don't think we need it now and I don't think it offers much advantage if we are merging `options` anyway. Let me know if I'm missing something!

Reply:
Technically `{ keys: Object.keys(rcConf) }` guarantee(we have test coverage on it) that we merge only the properties from  `rcResult.config` and all other properties from `options` will not be changed. For our particular use case it may be even more explicit `{ keys: ['reject'] }`, but current implementation is more generic and looks pretty safe. Logic behind `mergeOptions` is simple - it's the same `Object.assign` but if some props are arrays, then we merge them also. Additional  `keys` parameter allow us to be sure that may be changed only properties explicitly specified in `rcConfig` to prevent  unintended consequences.

Any case all this feature isolated from all other code with `--deep` and `--recursive` options, so this enhancement fix the problem and not brings the breaking changes.

Reply:
> Technically `{ keys: Object.keys(rcConf) }` guarantee(we have test coverage on it) that we merge only the properties from `rcResult.config` and all other properties from `options` will not be changed.

I'm not sure I understand. If an option does not exist in `rcResult`, then how could it overwrite `options`?

i.e. `_.pick(options2, Object.keys(options2))` is a noop

Reply:
Correct. If no any properties in `rcResult` then nothing to merge for this config, i.e. no need to override something in `options`. The idea is to override only properties, which explicitly defined in `rcResult`. If nothing defined, then nothing to do.

Reply:
Thanks for your response. I was unclear in my last message, so let me try again. Please correct me if I am missing something.

What I'm seeing is that on L515 you call:


```js
mergeOptions(options, rcConfig, { keys: Object.keys(rcConfig) })
```

And in merge-options [L15](https://github.com/raineorshine/npm-check-updates/pull/897/files#diff-c37a68c010b04a2f15f188f945826efc4a558a607f7d1871fda1d557ce4d25ffR15) we have: 

```js
options2 = opts.keys ? _.pick(options2 || {}, opts.keys) : options2 || {} 
```

So when the arguments get substituted through, we get:

```js
options2 = opts.keys ? _.pick(rcConfig, Object.keys(rcConfig) : rcConfig
```

Which is always `options2 = options2`, regardless of the inputs. Thus the `keys` logic is entirely superfluous which I don't believe was your intention. If you could remove `opts.keys` I think everything else looks good to go.